src/FactorAPI/LoginRequest.java

```diff
@@ -0,0 +1,19 @@
+package factorapi;
+//What a user needs to login and submit their request
+public class LoginRequest {
+    String username;
+    String encryptedPwd;
+    int[] factors;
+    String delimiter;
+    String source;
```

Comment:
Could create a new interface for source/destination, called StorageKey or something. This way, we can leave the type generic without committing to a single method of inputting and outputting data.

Reply:
So instead of String source, it would be StorageKey source or something along those lines.

Reply:
Yeah it would change to whatever is needed. The user just needs to put a source in for where they want it stored.